### ISSUE TITLE
Browser crash report sometimes raises exception: Converting circular structure to JSON

### ISSUE THREAD

**Issue by leigh-johnson** [bitsy-ai/printnanny-os#218]
## **Describe the bug**
Submitting a crash report **sometimes** results in an uncaught exception being raised:

```
logging.ts:29 TypeError: Converting circular structure to JSON
    --> starting at object with constructor 'Object'
    |     property 'subscriptions' -> object with constructor 'Object'
    |     property 'mux' -> object with constructor 'Object'
    --- property 'protocol' closes the circle
    at JSON.stringify (<anonymous>)
    at eB (logging.ts:34:11)
    at Proxy.sendCrashReport (alerts.ts:91:27)
    at Proxy.<anonymous> (pinia.mjs:1365:30)
    at l (CrashReportForm.vue:204:26)
    at vee-validate.esm.js:2025:32
console.error @ logging.ts:29
```

## **To Reproduce**
What are the steps to reproduce the issue?
1. Open PrintNanny Mission Control
2. Submit a crash report
3. Notice: *sometimes* there's an error in the console converting a circular structure to JSON

I have a feeling this is caused by logging pinia object references instead of the `toRaw` representation. 

## Issue.txt

```
-----------------------
Build Info:  |
-----------------------
IMAGE_NAME = printnanny-release-image-raspberrypi4-64-20230124200428
DISTRO_PRETTY = PrintNanny Linux 0.6.0 (Feldspar)
VARIANT_NAME = PrintNanny OS Core
VARIANT_ID = core
DISTRO = printnanny
DISTRO_NAME = PrintNanny Linux
DISTRO_VERSION = 0.6.0
DISTRO_CODENAME = Feldspar
ID_LIKE = BitsyLinux
HOME_URL = https://printnanny.ai
BUG_REPORT_URL = https://github.com/bitsy-ai/printnanny-os/issues
YOCTO_VERSION = 4.0.2
YOCTO_CODENAME = Langdale
MACHINE = raspberrypi4-64
TUNE_PKGARCH = cortexa72
-----------------------
Layer Revisions:      |
-----------------------
meta              = langdale:a3e3b740e140d036122f7b11e2ac452bda548444
meta-poky         = langdale:a3e3b740e140d036122f7b11e2ac452bda548444
meta-yocto-bsp    = langdale:a3e3b740e140d036122f7b11e2ac452bda548444
meta-raspberrypi  = libcamera-v0.0.3:104a8b9290ab4da92c8f693e2fdc42a6396975d2
meta-oe           = langdale:c5668905a6d8a78fb72c2cbf8b20e91e686ceb86
meta-python       = langdale:c5668905a6d8a78fb72c2cbf8b20e91e686ceb86
meta-multimedia   = langdale:c5668905a6d8a78fb72c2cbf8b20e91e686ceb86
meta-bitsy        = printnanny-snapshot-service:ae6710e5858ac0b13cd10c89ead66ba80537c6ae
meta-neural-network = master:fed68f0a659c9b6431985c156141dc28dc4c2275
meta-printnanny   = printnanny-snapshot-service:ae6710e5858ac0b13cd10c89ead66ba80537c6ae
meta-networking   = langdale:c5668905a6d8a78fb72c2cbf8b20e91e686ceb86
meta-filesystems  = langdale:c5668905a6d8a78fb72c2cbf8b20e91e686ceb86
meta-initramfs    = langdale:c5668905a6d8a78fb72c2cbf8b20e91e686ceb86
meta-webserver    = langdale:c5668905a6d8a78fb72c2cbf8b20e91e686ceb86
meta-swupdate     = master:5e611f44db1b9333342ed61d77a3c3ed7be69c2b
workspace         = langdale:a3e3b740e140d036122f7b11e2ac452bda548444
meta-microcontroller = master:9f396e595b2d639be7f2b494f4a61e48d9495058
```

**Comment by leigh-johnson** [comment]
I'm also seeing timeouts from the NATs message handler, which attempts to zip system logs and attach them to the crash report:

<img width="660" alt="Screenshot 2023-01-24 at 7 08 43 PM" src="https://user-images.githubusercontent.com/2601819/214471725-99dee9d0-dd63-4ac1-bc7b-7d1739c4e550.png">

```
logging.ts:29 Error enriching CrashReport with PrintNanny OS system logs NatsError: TIMEOUT
    at Ce.errorForCode (nats.js:80:16)
    at _f (nats.js:410:27)
    at new Pk (nats.js:1341:22)
    at Proxy.request (nats.js:9281:23)
    at Proxy.enrichCrashReport (alerts.ts:65:44)
    at async Proxy.sendCrashReport (alerts.ts:116:15)
    at async l (CrashReportForm.vue:204:14)
```

**Comment by leigh-johnson** [comment]
Two major improvements to the crash report system:

1. Switched to non-blocking async calls in most NATs message handler paths, especially the crash report zip builder
2. Using shallow object reads when logging any vue values to the console: https://vuejs.org/api/reactivity-advanced.html#shallowreadonly

**Comment by leigh-johnson** [comment]
I moved the base distro version from `langdale` to `mickledore` to pick up Rust >=1.64, `IntoFuture` API needed by the `async-nats` crate. I'm just taking note of this here, because usually, we'd want to do a distro version upgrade in a new release (not a patch release). 

The major changes are Python 3.10 -> Python 3.11, gcc 12.0 -> 12.2, and Rust 1.63 -> 1.66
ref: https://github.com/bitsy-ai/meta-bitsy/pull/390

**Comment by leigh-johnson** [comment]
:checkered_flag: Verified this is fixed in the latest v0.6.1 nightly.
